scripts/copyFL2

```diff
@@ -9,54 +9,76 @@
 # The offset value is bytes in decimal.
 # TODO - one day, this offset will be wrong.  (geteltorito could be taught
 # to output the starting sector it found as a starting point)
-FAT_OFFSET=71680
+FAT_OFFSET="${FAT_OFFSET:-71680}"
+
+# Allow multiple FL2 files to be present in a give fat16 filesystem as long as they are all the same (bitwise identical).
+# For some models like the x200 where sold with different SPI flash sizes (where the BIOS images is stored) with different BIOS "(anti)-features".
+# All of them will be patched.
+ALLOW_MULTIPLE_FL2_IF_SAME="${ALLOW_MULTIPLE_FL2_IF_SAME:-1}"
 
 DIR="$1"
 case "$DIR" in
     from_iso) ;;
     to_iso)   ;;
     *)
-        echo direction is either from_iso or to_iso
+        echo "Direction is either from_iso or to_iso" 2>&1
         exit 1
         ;;
 esac
 shift
 
 ISO="$1"
 if [ ! -e "$ISO" ]; then
-    echo iso file must exist
+    echo "iso file must exist" 2>&1
     exit 1
 fi
 shift
 
 FILENAME="$1"
 if [ -z "$FILENAME" ]; then
-    echo need filename
+    echo "Need filename" 2>&1
     exit 1
 fi
 shift
 
 FL2=$(mdir -i "$ISO"@@"$FAT_OFFSET" -/ -b |grep FL2)
 if [ -z "$FL2" ]; then
-    echo "Error: could not find any FL2 files in $ISO"
+    echo "Error: could not find any FL2 files in $ISO" 2>&1
     exit 1
 fi
-if [ $(echo "$FL2" |wc -w) -ne 1 ]; then
-    echo "Error: $ISO has more than one FL2 file:"
+if [ "$ALLOW_MULTIPLE_FL2_IF_SAME" -ne "1" ] && [ "$(echo "$FL2" |wc -w)" -ne 1 ]; then
+    echo "Error: $ISO has more than one FL2 file:" 2>&1
     echo "$FL2"
     exit 1
 fi
 
 from_iso() {
-    mcopy -i "$ISO"@@"$FAT_OFFSET" "$FL2" "$FILENAME"
+    if [ "$(echo "$FL2" |wc -w)" -eq 1 ]; then
+        mcopy -i "$ISO"@@"$FAT_OFFSET" "$FL2" "$FILENAME"
+    else
+        echo "$FL2" | while read fl2_filepath; do
+            mcopy -ni "$ISO"@@"$FAT_OFFSET" "$fl2_filepath" "$FILENAME"
+
+            if [ -z "${first_fl2_checksum:-}" ]; then
+                first_fl2_checksum="$(sha512sum "$FILENAME" | cut -d" " -f1)"
+            elif [ "$first_fl2_checksum" != "$(sha512sum "$FILENAME" | cut -d" " -f1)" ]; then
+                rm -f "$FILENAME"
+                echo "Error: $ISO has more than one FL2 file which are not identical:" 2>&1
+                echo "$FL2"
+                exit 1
+            fi
+        done
+    fi
```

Comment:
> Also, I have already found ISO files from Lenovo where they have included multiple

I guess there will always be this one case which breaks our assumption but requiring extra work for all normal cases where this can be handled automatically seems not right either.

> Finally, I'm trying to move towards a state where none of the inner names are hidden

Sounds good. They are not really hidden with this approach. Just that multiple FL2 files are updated when existing and started out equal. This can be made clear by using 'x' as you mentioned which I think is more transparent.

Ref: https://github.com/hamishcoleman/thinkpad-ec/commit/43506237f04e84557ed7b82f7822e11204411056#commitcomment-22825182